worldengine/common.py

```diff
@@ -120,3 +120,33 @@ def array_to_matrix(array, width, height):
         for x in range(width):
             matrix[y].append(array[y * width + x])
     return matrix
+
+def _equal(a, b):
+    #recursion on subclasses of types: tuple, list, dict
+    #specifically checks             : float, ndarray
```

Comment:
It is also used as World.operator== - I am not sure if there are any world-comparisons in the code outside of the tests. But I guess it is worth a try and the code can be moved, yes.